framework/app.mk

```diff
@@ -81,8 +100,13 @@ app_deps     := $(patsubst %.$(obj-suffix), %.$(obj-suffix).d, $(objects)) \
                 $(patsubst %.C, %.$(obj-suffix).d, $(main_src)) \
                 $(ADDITIONAL_APP_DEPS)
 
+app_test_deps     := $(patsubst %.$(obj-suffix), %.$(obj-suffix).d, $(test_objects)) \
+                $(patsubst %.c, %.$(obj-suffix).d, $(test_csrcfiles))
 depend_dirs := $(foreach i, $(DEPEND_MODULES), $(MOOSE_DIR)/modules/$(i)/include)
 depend_dirs += $(APPLICATION_DIR)/include
+ifneq ($(wildcard $(APPLICATION_DIR)/test/include/*),)
```

Comment:
This is just a test to see if there is anything in `test/include` and if there is then add `test/include` to the list of directories that we use with "-I". I guess it doesn't hurt to add the directory regardless.